src/app/backend/integration/metric/common/common.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+  "github.com/kubernetes/dashboard/src/app/backend/api"
+  metricapi "github.com/kubernetes/dashboard/src/app/backend/integration/metric/api"
+)
+
+type CommonSelector struct {
```

Comment:
Heapster and sidecar selectors are similar only because sidecar was built to mimic heapster API. This will not always be true, that is why logic was split. Additionally, heapster integration will be soon removed completely from our code.